airflow/template/templater.py

```diff
@@ -0,0 +1,187 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Collection, Iterable, Sequence
+
+from airflow.utils.context import Context
+from airflow.utils.helpers import render_template_as_native, render_template_to_string
+from airflow.utils.log.logging_mixin import LoggingMixin
+from airflow.utils.mixins import ResolveMixin
+from airflow.utils.session import NEW_SESSION, provide_session
+
+if TYPE_CHECKING:
+    import jinja2
+    from sqlalchemy.orm import Session
+
+    from airflow import DAG
+
+
+class Templater(LoggingMixin):
```

Comment:
Not quite sure about this Templater name but don’t have a suggestion either. Perhaps add a `:meta private:` so we can rename this freely if we want to at any point?